From e7b290d9b42f0f3a7ceb9e32638aac30d827d797 Mon Sep 17 00:00:00 2001
From: Luc Verhaegen <libv@skynet.be>
Date: Thu, 1 Nov 2012 21:16:21 +0100
Subject: [PATCH 913/944] video:sunxi:disp: improve loop in Fb_Exit

Don't bother trying to unregister and release a framebuffer that is
NULL.

Signed-off-by: Luc Verhaegen <libv@skynet.be>
---
 drivers/video/sunxi/disp/dev_fb.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/video/sunxi/disp/dev_fb.c b/drivers/video/sunxi/disp/dev_fb.c
index 739b2c6..eace6e8 100644
--- a/drivers/video/sunxi/disp/dev_fb.c
+++ b/drivers/video/sunxi/disp/dev_fb.c
@@ -1671,8 +1671,10 @@ __s32 Fb_Exit(void)
 	__u8 fb_id = 0;
 
 	for (fb_id = 0; fb_id < SUNXI_MAX_FB; fb_id++) {
-		if (g_fbi.fbinfo[fb_id] != NULL)
-			Display_Fb_Release(FBIDTOHAND(fb_id));
+		if (g_fbi.fbinfo[fb_id] == NULL)
+			continue;
+
+		Display_Fb_Release(FBIDTOHAND(fb_id));
 
 		unregister_framebuffer(g_fbi.fbinfo[fb_id]);
 		framebuffer_release(g_fbi.fbinfo[fb_id]);
-- 
1.8.0

